### PR TITLE
[XPU] Enable NVIDIA-Nemotron-3-Nano-30B-A3B-BF16 on Intel XPU backend

### PR DESCRIPTION
**Pull request by jmunetong**
# [XPU] Enable NVIDIA-Nemotron-3-Nano-30B-A3B-BF16 on Intel XPU

## Summary

Enable `nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16` (hybrid Mamba + attention
MoE, squared-ReLU activation) on Intel XPU with the `intel_xpu` attention
backend at `tp=4` on 4× Intel Arc Pro B60. Before this PR the model crashes at
load on XPU: the `causal_conv1d` import gate only binds on CUDA/NPU, and the MoE
path rejects the `relu2` activation Nemotron-H uses.

No CUDA/NPU behavior changes — both edits are additive XPU-only branches.

**Key changes:**

- Bind `causal_conv1d_*` to the pure-Triton implementation on XPU
- Admit `relu2` MoE activation on XPU

---

## GSM8K Accuracy Results (lm-eval, limit=250, thinking ON)

| Filter | XPU tp=4 B60 | B200 BF16 (cookbook §5.2.1) | Δ |
|---|---:|---:|---:|
| `flexible-extract` exact_match | **0.584 ± 0.031** | 0.560 | +0.024 |
| `strict-match` exact_match | **0.856 ± 0.022** | 0.845 | +0.011 |

XPU matches or exceeds the B200 BF16 cookbook reference within stderr.

---

## Reproducing

```bash
# Launch server (tp=4, 4× XPU tiles)
ZE_AFFINITY_MASK=0,1,2,3 SGLANG_USE_SGL_XPU=1 \
python3 -m sglang.launch_server \
  --model-path nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16 \
  --host 127.0.0.1 --port 30000 \
  --device xpu --tp 4 --trust-remote-code \
  --disable-overlap-schedule --disable-radix-cache --page-size 64 \
  --attention-backend intel_xpu --model-impl sglang \
  --tool-call-parser qwen3_coder --reasoning-parser nemotron_3 \
  --mem-fraction-static 0.85 --context-length 8192 \
  --chunked-prefill-size 1024 --max-running-requests 8 --cuda-graph-max-bs 8

# GSM8K
lm_eval --model local-completions --tasks gsm8k \
  --model_args "model=nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16,base_url=http://127.0.0.1:30000/v1/completions,num_concurrent=4,max_retries=3,tokenized_requests=False,max_lengths=16384" \
  --gen_kwargs '{"chat_template_kwargs":{"thinking":true}}' \
  --batch_size 32 --limit 250
```

---

## Modifications

| # | File | Change |
|---|---|---|
| 1 | `python/sglang/srt/layers/attention/mamba/mamba.py` (+17) | Import `is_xpu`; add an `elif is_xpu():` branch to the `causal_conv1d_*` import gate that binds `causal_conv1d_fn`, `causal_conv1d_fn_triton`, `causal_conv1d_update`, and `causal_conv1d_update_triton` to the portable Triton implementation (no native XPU kernel yet). |
| 2 | `python/sglang/srt/layers/quantization/unquant.py` (+1) | Add `"relu2"` to the `UnquantizedFusedMoEMethod.forward_xpu` activation whitelist (Nemotron-H uses squared-ReLU); CUDA already handles `Relu2`. |
| 3 | `.gitignore` (+5) | Ignore the local-only Nemotron-3 XPU test (`tp=4`, too large for the 1-GPU XPU CI runner). |

---

## Tests

- Verified locally with an E2E smoke test on `--attention-backend intel_xpu --tp 4` (single Q&A, returns coherent output). The test is kept local-only (gitignored) because it requires tp=4 and no 4-GPU XPU CI runner exists.
- GSM8K accuracy verified (tp=4, B60) within stderr of the B200 BF16 cookbook baseline (table above).

---

## Checklist

- [x] Pre-commit clean
- [x] XPU smoke test passing (local, tp=4)
- [x] GSM8K accuracy verified (no regression vs B200 BF16 cookbook)
- [ ] Runtime performance / profiling (to follow in a later update)

---
### CI States

Latest PR Test (Base): :x: [Run #27154611547](https://github.com/sgl-project/sglang/actions/runs/27154611547)
Latest PR Test (Extra): :x: [Run #27154611141](https://github.com/sgl-project/sglang/actions/runs/27154611141)

### CODE REVIEW

File: test/srt/xpu/test_nvidia_nemotron_3_nano.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+Run from test/srt::
+
+  python3 -m unittest xpu.test_nvidia_nemotron_3_nano.TestNemotron3Nano30BXPU.test_simple_code_qa
+
+Appends to ``nemotron_3_nano_30b_comparison.txt`` in this directory.
+
+Server is started with ``sglang serve`` (``--model-impl sglang``).
+"""
+
```
Comment (mingfeima) [inline test/srt/xpu/test_nvidia_nemotron_3_nano.py]: need something to do the register `register_xpu_ci(est_time=xxx, suite="stage-b-test-1-gpu-xpu")`

xxx - the runtime needed for the case, don't need to be 100% accurate, just an estimate will be fine.

this is the weight used to do test case partitions for multiple jobs.

File: python/sglang/srt/layers/attention/mamba/mamba.py [hunk 2]
```diff
@@ -56,6 +57,22 @@
...
+    )
+    from sglang.srt.layers.attention.mamba.causal_conv1d_triton import (
+        causal_conv1d_fn as causal_conv1d_fn_triton,
+    )
+    from sglang.srt.layers.attention.mamba.causal_conv1d_triton import (
+        causal_conv1d_update as causal_conv1d_update,
+    )
+    from sglang.srt.layers.attention.mamba.causal_conv1d_triton import (
+        causal_conv1d_update as causal_conv1d_update_triton,
+    )
```
Comment (mingfeima) [inline python/sglang/srt/layers/attention/mamba/mamba.py:75]: @Xia-Weiwen how we run qwen3.5 on xpu for this part? do we really need this change?

**Review by mingfeima — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/attention/mamba/mamba.py (hunk 2 above)
Comment (Xia-Weiwen) [reply python/sglang/srt/layers/attention/mamba/mamba.py:75]: Qwen3.5 uses the same triton kernel but it does not run into this file. It runs into https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/layers/attention/linear/gdn_backend.py

File: test/srt/xpu/test_nvidia_nemotron_3_nano.py (hunk 1 above)
Comment (Xia-Weiwen) [reply test/srt/xpu/test_nvidia_nemotron_3_nano.py]: Example: https://github.com/sgl-project/sglang/blob/main/test/registered/attention/test_chunk_gated_delta_rule.py

File: python/sglang/srt/layers/attention/mamba/mamba.py (hunk 2 above)
Comment (mingfeima) [reply python/sglang/srt/layers/attention/mamba/mamba.py:75]: so on the cuda side, it has been diverged?

Comment (Xia-Weiwen) [reply python/sglang/srt/layers/attention/mamba/mamba.py:75]: Strictly speaking, mamba != GDN, so it is not surprising if they diverge but I didn't study the two paths so I might be wrong about the details in SGLang. :)

**Comment by jmunetong** [comment]
**Note: XPU Nemotron-3 Nano test omitted from CI**

The end-to-end XPU test for `nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16` (`test/registered/xpu/test_nvidia_nemotron_3_nano.py`) has been removed from version control and added to `.gitignore`.

The model is too large to run in CI: it requires `--tp=4` (30B-A3B hybrid Mamba + MoE), and there is no 4-GPU XPU CI runner. The test is kept on disk for local development only and is intentionally never tracked or pushed.

It has been validated locally on a 4-XPU host (passes with `ZE_AFFINITY_MASK=0,1,2,3`).

File: test/srt/xpu/test_nvidia_nemotron_3_nano.py (hunk 1 above)
Comment (jmunetong) [reply test/srt/xpu/test_nvidia_nemotron_3_nano.py]: Would it be possible to skip the CI test for this model underscores the following memory requirements during CI:
 - Model weights: ~63.2 GB (BF16), across 13 safetensors shards — this is the authoritative figure from the index (total_size =
  63,155,886,464 bytes ≈ 58.8 GiB / 63.2 GB).
  - Total cache dir: 59 GiB (~63 GB) including config/tokenizer (weights dominate).

If we are able to skip this test, then this should be ready for merging.

**Comment by jmunetong** [comment]
@mingfeima Pr https://github.com/sgl-project/sgl-kernel-xpu/pull/234 from sgl-kernel-xpu enables support for fp32 gating logits which did not allow this model to run. Now, it should be ready.

**Review by mingfeima — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]
